flow/controllers/imitation_learning/imitating_network.py

```diff
@@ -0,0 +1,267 @@
+
+        """
+        observation = env.get_state()
+        return self.get_accel_from_observation(observation)
+
+
+    def add_to_replay_buffer(self, rollout_list):
+        """
+        Add data to a replay buffer
+
+        Parameters
+        ----------
+        rollout_list : list
+            list of rollout dictionaries
+        """
+
+        self.replay_buffer.add_rollouts(rollout_list)
+
+
+    def sample_data(self, batch_size):
+        """
+        Sample a batch of data from replay buffer.
+
+        Parameters
+        ----------
+        batch_size : int
+            size of batch to sample
+        """
+
+        return self.replay_buffer.sample_batch(batch_size)
+
+    def save_network(self, save_path):
+        """
+        Save imitation network as a h5 file in save_path
+
+        Parameters
+        ----------
+        save_path : String
+            path to h5 file to save to
+        """
+
+        self.model.save(save_path)
+        # tensorboard
+
+        # writer = tf.summary.FileWriter('./graphs2', tf.get_default_graph())
+
+    def load_network(self, load_path):
+        """
+        Load imitation network from a h5 file in load_path
+
+        Parameters
+        ----------
+        load_path : String
+            path to h5 file containing model to load from
+        """
+        if self.stochastic:
+            self.model = tf.keras.models.load_model(load_path, custom_objects={'nll_loss': negative_log_likelihood_loss(self.variance_regularizer)})
+        else:
+            self.model = tf.keras.models.load_model(load_path)
+
+
+    def save_network_PPO(self, save_path):
+        """
+        Build a model, with same policy architecture as imitation network, to run PPO, copy weights from imitation, and save this model.
+
+        Parameters
+        ----------
+        load_path : save_path
+            path to h5 file to save to
+        """
+
+        input = tf.keras.layers.Input(self.model.input.shape[1].value)
+        curr_layer = input
+
+        # number of hidden layers
+        num_layers = len(self.model.layers) - 2
+
+        # build layers for policy
+        for i in range(num_layers):
+            size = self.model.layers[i + 1].output.shape[1].value
+            activation = tf.keras.activations.serialize(self.model.layers[i + 1].activation)
+            curr_layer = tf.keras.layers.Dense(size, activation=activation, name="policy_hidden_layer_{}".format(i + 1))(curr_layer)
+        output_layer_policy = tf.keras.layers.Dense(self.model.output.shape[1].value, activation=None, name="policy_output_layer")(curr_layer)
+
+        # build layers for value function
+        curr_layer = input
+        for i in range(num_layers):
+            size = self.fcnet_hiddens[i]
+            curr_layer = tf.keras.layers.Dense(size, activation="tanh", name="vf_hidden_layer_{}".format(i+1))(curr_layer)
+        output_layer_vf = tf.keras.layers.Dense(1, activation=None, name="vf_output_layer")(curr_layer)
```

Comment:
watch out; you're implicitly assuming that vf_share_layers is never true. This should be warned about somewhere